docs/source/data/fetching-rest.mdx

```diff
@@ -0,0 +1,421 @@
+import { RESTDataSource } from '@apollo/datasource-rest';
+
+class MoviesAPI extends RESTDataSource {
+  override baseURL = 'https://movies-api.example.com/';
+
+  async getMovie(id):Promise<Movie> {
+    return this.get<Movie>(`movies/${encodeURIComponent(id)}`);
+  }
+
+  async getMostViewedMovies(limit = '10'):Promise<Movie[]> {
+    const data = await this.get('movies', {
+      params: {
+        per_page: limit,
+        order_by: 'most_viewed',
+      },
+    });
+    return data.results;
+  }
+}
+```
+
+</MultiCodeBlock>
+
+You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods should use the built-in convenience methods (e.g., `get` and `post`) to perform HTTP requests, helping you add query parameters, parse and cache JSON results, dedupe requests, and handle errors.
+
+## Adding data sources to Apollo Server's context
+
+You can add data sources to the `context` initialization function, like so:
+
+<MultiCodeBlock>
+
+```ts title="index.ts"
+//highlight-start
+interface ContextValue {
+  dataSources: {
+    moviesAPI: MoviesAPI;
+    personalizationAPI: PersonalizationAPI;
+  };
+}
+//highlight-end
+
+const server = new ApolloServer<ContextValue>({
+  typeDefs,
+  resolvers,
+});
+
+const { url } = await startStandaloneServer(server, {
+  context: async () => {
+    const { cache } = server; // highlight-line
+    return {
+      // We create new instances of our data sources with each request,
+      // passing in our server's cache.
+      //highlight-start
+      dataSources: {
+        moviesAPI: new MoviesAPI({ cache }),
+        personalizationAPI: new PersonalizationAPI({ cache }),
+      },
+      //highlight-end
+    };
+  },
+});
+
+console.log(`🚀  Server ready at ${url}`);
+```
+
+</MultiCodeBlock>
+
+<!-- TODO(AS4) add link to context article once exists  -->
+Apollo Server calls [the `context` initialization](../resolvers/#the-context-argument) function for _every incoming operation_. This means:
+- For every operation, `context` returns an _object_ containing new instances of your `RESTDataSource` subclasses (in this case, `MoviesAPI` and `PersonalizationAPI`).
+- The **`context` function should create a new instance of each `RESTDataSource` subclass for each operation.** If multiple operations share a single data source instance, you might accidentally combine results from multiple operations.
+
+Your resolvers can then access your data sources from the shared `context` object and use them to fetch data:
+
+```ts title="resolvers.ts"
+const resolvers = {
+  Query: {
+    movie: async (_, { id }, { dataSources }) => {
+      return dataSources.moviesAPI.getMovie(id);
+    },
+    mostViewedMovies: async (_, __, { dataSources }) => {
+      return dataSources.moviesAPI.getMostViewedMovies();
+    },
+    favorites: async (_, __, { dataSources }) => {
+      return dataSources.personalizationAPI.getFavorites();
+    },
+  },
+};
+```
+
+## Caching
```

Comment:
it is probably important to mention that this cache respects headers like cache-control from the API and doesn't cache things that the backend doesn't say to cache!

Reply:
Great point; I added a sentence above telling folks to enable caching:

> If you've set up [caching](../preformance/caching/#recommended-starting-usage) for your server, a data source can help store the results of past fetches for particular fields in your schema.

As well as added a note after the first code snippet:

> Note that `RESTDataSource` only caches data for the fields specified by your server. To learn more about how to set up caching for your server, see [Server-side caching](../preformance/caching).

Reply:
That's not really it. It's that it will cache data that the *REST API* specifies as cacheable in its HTTP response headers. This has nothing to do with fields or resolvers or schema or directives, because RESTDataSource is a caching HTTP client that has literally nothing to do with GraphQL or Apollo Server except for the fact that it is one way of solving a problem that many Apollo Server users do have.

Reply:
Okay, I obviously did not understand how that worked; thank you for explaining that! 

How does this sentence feel instead:

> The `RESTDataSource` class can cache results if the REST API it fetches from specifies caching headers in its HTTP responses (e.g., [`cache-control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control)).

Reply:
You probably didn't understand it because we never bothered to describe what the entire point of this project is anywhere!

That sentence looks great.